config/overrides.go

```diff
@@ -0,0 +1,172 @@
+
+// GroupMap contains all overrides we'd like to make to the default group search.
+var GroupMap = map[string]GroupKindCalculator{
+	"rancher2_app":                         ReplaceGroupWords("app", 0),
+	"rancher2_app_v2":                      ReplaceGroupWords("appv2", 0),
+	"rancher2_auth_config_activedirectory": ReplaceGroupWords("authconfig", 2),
+	"rancher2_auth_config_adfs":            ReplaceGroupWords("authconfig", 2),
+	"rancher2_auth_config_azuread":         ReplaceGroupWords("authconfig", 2),
+	"rancher2_auth_config_freeipa":         ReplaceGroupWords("authconfig", 2),
+	"rancher2_auth_config_github":          ReplaceGroupWords("authconfig", 2),
+	"rancher2_auth_config_keycloak":        ReplaceGroupWords("authconfig", 2),
+	"rancher2_auth_config_okta":            ReplaceGroupWords("authconfig", 2),
+	"rancher2_auth_config_openldap":        ReplaceGroupWords("authconfig", 2),
+	"rancher2_auth_config_ping":            ReplaceGroupWords("authconfig", 2),
+	"rancher2_catalog":                     ReplaceGroupWords("catalog", 0),
+	"rancher2_catalog_v2":                  ReplaceGroupWords("catalogv2", 0),
+	"rancher2_certificate":                 ReplaceGroupWords("project", 0),
+	"rancher2_cloud_credential":            ReplaceGroupWords("", 0),
+	"rancher2_cluster":                     ReplaceGroupWords("cluster", 0),
+	"rancher2_cluster_v2":                  ReplaceGroupWords("clusterv2", 0),
+	"rancher2_config_map_v2":               ReplaceGroupWords("cluster", 0),
+	"rancher2_etcd_backup":                 ReplaceGroupWords("cluster", 0),
+	"rancher2_feature":                     ReplaceGroupWords("", 0),
+	"rancher2_machine_config_v2":           ReplaceGroupWords("", 0),
+	"rancher2_multi_cluster_app":           ReplaceGroupWords("appv2", 0),
+	"rancher2_namespace":                   ReplaceGroupWords("project", 0),
+	"rancher2_node_driver":                 ReplaceGroupWords("node", 0),
+	"rancher2_node_pool":                   ReplaceGroupWords("node", 0),
+	"rancher2_node_template":               ReplaceGroupWords("node", 0),
+	"rancher2_notifier":                    ReplaceGroupWords("cluster", 0),
+	"rancher2_project":                     ReplaceGroupWords("project", 0),
+	"rancher2_registry":                    ReplaceGroupWords("project", 0),
+	"rancher2_role_template":               ReplaceGroupWords("", 0),
+	"rancher2_secret":                      ReplaceGroupWords("secret", 0),
+	"rancher2_secret_v2":                   ReplaceGroupWords("secretv2", 0),
+	"rancher2_setting":                     ReplaceGroupWords("", 0),
+	"rancher2_storage_class_v2":            ReplaceGroupWords("cluster", 0),
+	"rancher2_token":                       ReplaceGroupWords("admin", 0),
+	"rancher2_user":                        ReplaceGroupWords("", 0),
+}
+
+// ReplaceGroupWords uses given group as the group of the resource and removes
+// a number of words in resource name before calculating the kind of the resource.
```

Comment:
could you provide an example of transformation in the comment from the map above? It is not very obvious how replacing works without getting into context

Reply:
It is the same replacement logic used by aws and gcp jet providers:
https://github.com/crossplane-contrib/provider-jet-aws/blob/main/config/overrides.go#L275
https://github.com/crossplane-contrib/provider-jet-gcp/blob/main/config/overrides.go#L102

Given a terraform resource name, e.g. rancher2_app it strips off the common provider prefixes (rancher2). The remaining first x words (_ as seperator) are dropped (x is the number in the func call). The remaining words are combined to form the Kind, the group is set with the first func arg.
So `rancher2_app` becomes `App.app.rancher.jet...` instead of `App.rancher2.rancher.jet....`
and `rancher2_app_v2` becomes `App.appv2.rancher.jet...` instead of `V2.app.rancher.jet...`

Grouping idea is based off the v2 grouping used by the aws provider, aws_elb -> group elb and aws_alb -> group elbv2.
If we are oke with our own v1/v2 resource scheme, I would opt for `App.app.rancher.jet...` and `AppV2.app.rancher.jet...`. The app group only has 2 resources, but if you look at cluster resources, it has more resources in the cluster group. Atm `Cluster.clusterv2.rancher.jet` is the only resource in the clusterv2 group, but in the cluster group there are resources (except Cluster.cluster.rancher.jet obviously) that interact with a clusterv2, as those resources in terraform just take a cluster id, not caring if you got that cluster id from a rancher2_cluster of rancher_cluster_v2 resource.